tests/components/xiaomi_miio/test_vacuum.py

```diff
@@ -176,46 +176,53 @@ def mirobo_is_on_fixture():
     mock_vacuum.status().state_code = 5
     mock_vacuum.dnd_status().enabled = False
 
-    mock_timer_1 = mock.MagicMock()
+    mock_timer_1 = MagicMock()
     mock_timer_1.enabled = True
     mock_timer_1.cron = "5 5 1 8 1"
     mock_timer_1.next_schedule = datetime(2020, 5, 23, 13, 21, 10, tzinfo=utc)
 
-    mock_timer_2 = mock.MagicMock()
+    mock_timer_2 = MagicMock()
     mock_timer_2.enabled = False
     mock_timer_2.cron = "5 5 1 8 2"
     mock_timer_2.next_schedule = datetime(2020, 5, 23, 13, 21, 10, tzinfo=utc)
 
     mock_vacuum.timer.return_value = [mock_timer_1, mock_timer_2]
 
-    with mock.patch(
-        "homeassistant.components.xiaomi_miio.vacuum.Vacuum"
-    ) as mock_vaccum_cls:
-        mock_vaccum_cls.return_value = mock_vacuum
-        yield mock_vacuum
-
-
-@pytest.fixture(name="mock_mirobo_errors")
-def mirobo_errors_fixture():
-    """Mock mock_mirobo_errors to simulate a bad vacuum status request."""
-    mock_vacuum = mock.MagicMock()
-    mock_vacuum.status.side_effect = OSError()
-    with mock.patch(
-        "homeassistant.components.xiaomi_miio.vacuum.Vacuum"
-    ) as mock_vaccum_cls:
+    with patch("homeassistant.components.xiaomi_miio.vacuum.Vacuum") as mock_vaccum_cls:
         mock_vaccum_cls.return_value = mock_vacuum
         yield mock_vacuum
 
 
-async def test_xiaomi_exceptions(hass, caplog, mock_mirobo_errors):
-    """Test vacuum supported features."""
+async def test_xiaomi_exceptions(hass, caplog, mock_mirobo_is_on):
+    """Test error logging on exceptions."""
     entity_name = "test_vacuum_cleaner_error"
-    await setup_component(hass, entity_name)
+    entity_id = await setup_component(hass, entity_name)
+
+    def is_available():
+        state = hass.states.get(entity_id)
+        return state.state != STATE_UNAVAILABLE
 
+    # The initial setup has to be done successfully
     assert "Initializing with host 192.168.1.100 (token 12345...)" in caplog.text
-    assert mock_mirobo_errors.status.call_count == 1
-    assert "ERROR" in caplog.text
-    assert "Got OSError while fetching the state" in caplog.text
+    assert "WARNING" not in caplog.text
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6372789)